src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,122 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
     @Override
     public void put(K key, V value) {
-
+        if (table == null || size == 0) {
+            resize();
+        }
+        int index = hash(key) % table.length;
+        if (table[index] != null) {
+            Node<K, V> currentNode = table[index];
+            while (currentNode != null) {
+                if (currentNode.key == key
+                        || (currentNode.key != null && currentNode.key.equals(key))) {
+                    currentNode.value = value;
+                    break;
+                }
+                if (currentNode.next == null) {
+                    currentNode.next = new Node<>(hash(key), key, value, null);
+                    size++;
+                    break;
+                }
+                currentNode = currentNode.next;
+            }
+        } else {
+            table[index] = new Node<>(hash(key), key, value, null);
+            size++;
+        }
+        if (size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (table != null) {
+            int index = hash(key) % table.length;
+            if (table[index] != null) {
+                Node<K, V> currentNode = table[index];
+                while (currentNode != null) {
+                    if (currentNode.key == key
+                            || (currentNode.key != null && currentNode.key.equals(key))) {
+                        return currentNode.value;
+                    }
+                    currentNode = currentNode.next;
+                }
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static int hash(Object key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode());
+    }
```

Comment:
1) why static?
2) Why Object?
3) Rework this method to provide bucket index instead of just hash
```suggestion
    private int getBucketIndex(K key) {
        return (key == null) ? 0 : Math.abs(key.hashCode()) % table.length;
    }
```